noise/conn.go

```diff
@@ -319,6 +320,9 @@ ContinueHandshake:
 			if !c.config.PublicKeyVerifier(hs.rs.PublicKey[:], receivedPayload) {
 				return errors.New("Noise: the received public key could not be authenticated")
 			}
+			c.isRemoteAuthenticated = true
```

Comment:
good point, I was toying with this idea of having a `isRemoteAuthenticated` value you can check.

I'm not sure it is useful as the handshake you're using should make it pretty clear that you're authenticating the remote, but I thought it could be "useful" as a defense-in-depth mechanism. What do you think?

It should also probably be exported if it's meant to be used by developers.

Reply:
I put it there because I used the `IsAuthenticated` method in `StaticKey`. If it's not really needed (see comment below) then we can totally drop that.
In my opinion, the semantic of `IsAuthenticated` is a bit vague or redundant: do you mean authenticated with a longterm static key ? Because users of that library must intentionally declare the pattern they want to use and if the handshake succeeds and a static key must have been authenticated in some way,  then it is by default `Authenticated` no ?

Reply:
> authenticated with a longterm static key?

not necessarily, I am trying to get back into the state of mind I had when I wrote this. But I think I wanted to create an out-of-band authentication. For example something like Noise_NN where you would extract a fingerprint and authenticate the connection post-handshake. Noise_NNoob maybe? Although I'm not sure it deserves a name. By having this `isRemoteAuthenticated` you can check if peers have actually verified the fingerprint.